src/Microsoft.AspNet.Http.Features/ISession.cs

```diff
@@ -1,20 +1,20 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using System;
 using System.Collections.Generic;
+using System.Threading.Tasks;
 
 namespace Microsoft.AspNet.Http.Features
 {
     public interface ISession
     {
-        void Load();
+        Task LoadAsync();
```

Comment:
But shouldn't that state be internal to the session implementations itself?...like from middleware we just commit and the implementation internally decides if it needs to commit or not...